src/Components/Web/src/Forms/InputRadio.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.AspNetCore.Components.Rendering;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    /// <summary>
+    /// An input component used for selecting a value from a group of choices.
+    /// </summary>
+    public class InputRadio<TValue> : InputBase<TValue>
+    {
+        /// <summary>
+        /// Gets the name of this <see cref="InputRadio{TValue}"/> group.
+        /// </summary>
+        protected string? GroupName { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the value that will be bound when this radio input is selected.
+        /// </summary>
+        [AllowNull]
+        [MaybeNull]
+        [Parameter]
+        public TValue SelectedValue { get; set; } = default;
+
+        /// <summary>
+        /// Gets or sets group name inherited from an ancestor <see cref="InputRadioGroup"/>.
+        /// </summary>
+        [CascadingParameter] InputRadioGroup? CascadedRadioGroup { get; set; }
+
+        /// <inheritdoc />
+        protected override void OnParametersSet()
+        {
+            GroupName = AdditionalAttributes != null && AdditionalAttributes.TryGetValue("name", out var nameAttribute) ?
+                Convert.ToString(nameAttribute) :
```

Comment:
Shouldn't we throw here if name attribute != 'string'? I imagine convert will throw if I you give it a random object it doesn't support, but it is worth considering.

Reply:
I would consider throwing directly here if the name is not a string, since GroupName requires a string